src/plugins/discover/public/application/components/sidebar/discover_field.tsx

```diff
@@ -108,78 +95,114 @@ export function DiscoverField({
-        tabIndex={0}
-        onClick={() => onShowDetails(!showDetails, field)}
-        onKeyPress={() => onShowDetails(!showDetails, field)}
-        data-test-subj={`field-${field.name}-showDetails`}
-      >
-        <span className="dscSidebarField__fieldIcon">
-          <FieldIcon
-            type={field.type}
-            label={getFieldTypeName(field.type)}
-            scripted={field.scripted}
+      <EuiPopover
+        ownFocus
+        display="block"
+        button={
+          <FieldButton
+            size="s"
+            className="dscSidebarItem"
+            isActive={infoIsOpen}
+            onClick={() => {
+              togglePopover();
+            }}
+            buttonProps={{ 'data-test-subj': `field-${field.name}-showDetails` }}
+            fieldIcon={dscFieldIcon}
+            fieldAction={actionButton}
+            fieldName={fieldName}
           />
-        </span>
-        <span
-          data-test-subj={`field-${field.name}`}
-          title={field.name}
-          className="dscSidebarField__name"
-        >
-          {useShortDots ? wrapOnDot(shortenDottedString(field.name)) : wrapOnDot(field.displayName)}
-        </span>
-        <span>
-          {field.name !== '_source' && !selected && (
-            <EuiButton
-              fill
-              size="s"
-              className="dscSidebarItem__action"
-              onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-                ev.preventDefault();
-                ev.stopPropagation();
-                toggleDisplay(field);
-              }}
-              data-test-subj={`fieldToggle-${field.name}`}
-              arial-label={addLabelAria}
-            >
-              {addLabel}
-            </EuiButton>
-          )}
-          {field.name !== '_source' && selected && (
-            <EuiButton
-              color="danger"
-              className="dscSidebarItem__action"
-              onClick={(ev: React.MouseEvent<HTMLButtonElement>) => {
-                ev.preventDefault();
-                ev.stopPropagation();
-                toggleDisplay(field);
-              }}
-              data-test-subj={`fieldToggle-${field.name}`}
-              arial-label={removeLabelAria}
-            >
-              {removeLabel}
-            </EuiButton>
-          )}
-        </span>
-      </div>
-      {showDetails && (
+        }
+        isOpen={infoIsOpen}
+        closePopover={() => setOpen(false)}
+        anchorPosition="rightUp"
+        panelClassName="dscSidebarItem__fieldPopoverPanel"
+      >
+        <EuiPopoverTitle>
+          {' '}
+          {i18n.translate('discover.fieldChooser.discoverField.fieldTopValuesLabel', {
+            defaultMessage: 'Top 5 values',
+          })}
+        </EuiPopoverTitle>
         <DiscoverFieldDetails
           indexPattern={indexPattern}
           field={field}
           details={getDetails(field)}
           onAddFilter={onAddFilter}
         />
```

Comment:
Hopefully most of the tests should be fine once this is fixed: #75025
The failing `getDetails` also contains lot's of legacy code, and with the new implementation it's executed with every field, so I'd suggest to handle it in a similar way like before, only when the `DiscoverFieldDetails` are displayed:
```suggestion
        {infoIsOpen && (
          <DiscoverFieldDetails
            indexPattern={indexPattern}
            field={field}
            details={getDetails(field)}
            onAddFilter={onAddFilter}
          />
        )}
```